[Search Application] [Preview] Add configuration button with Nav items

saarikabhasi opened the pull request:
## Summary

* Add configuration Button and comment out changes for View this API call - see [thread](https://elastic.slack.com/archives/C02U50QNEAG/p1682089775653529?thread_ts=1681758090.657279&cid=C02U50QNEAG) 

### Screen Recording




https://user-images.githubusercontent.com/55930906/234107309-40542d5f-253e-4e5b-8c2c-418530446e72.mov

saarikabhasi commented on x-pack/plugins/enterprise_search/public/applications/enterprise_search_content/components/engine/engine_search_preview/engine_search_preview.tsx:
```diff
@@ -64,7 +83,7 @@ class InternalEngineTransporter implements Transporter {
     });
 
-    this.setLastAPICall({ request, response });
+    // this.setLastAPICall({ request, response }); Uncomment when view this API call is needed
```
Commented implementation of view this API for 8.8

saarikabhasi commented:
@elasticmachine merge upstream

julianrosado commented:
@saarikabhasi Feedback 👇

1. Can we have no padding in the contextmenu, so it looks more like the first example here: https://eui.elastic.co/#/navigation/context-menu

![CleanShot 2023-04-24 at 13 23 36](https://user-images.githubusercontent.com/4490333/234070903-863e9a91-cbfb-4207-8c62-a4d74fe73554.png)

2. Can we also check the spacing on the individual menu items? Seems like theres some additional padding there?
![CleanShot 2023-04-24 at 13 24 07](https://user-images.githubusercontent.com/4490333/234071117-9c3dc144-6923-494d-9138-482c3fdd9456.png)
vs
![CleanShot 2023-04-24 at 13 24 29](https://user-images.githubusercontent.com/4490333/234071138-261b72ec-0570-4141-aef1-384a3f7aab61.png)

3. Can we make "Delete This App" red?

saarikabhasi commented:
With adjust padding size:
 
<img width="429" alt="Context menu" src="https://user-images.githubusercontent.com/55930906/234086725-112db638-4af8-4145-a389-633351a7e12f.png">

cc: @julianrosado

julianrosado commented:
@saarikabhasi It looks like "Settings" has different padding than "Content" and "Connect"? Can those two have the same padding as "Settings"?

saarikabhasi commented:
Updated with Adding new links for indices and schema tabs

joemcelroy commented on x-pack/plugins/enterprise_search/public/applications/enterprise_search_content/components/engine/engine_search_preview/engine_search_preview.tsx:
```diff
@@ -87,20 +109,179 @@ class InternalEngineTransporter implements Transporter {
+            iconSide="right"
+            onClick={setCloseConfiguration}
+          >
+            Configuration
```
not localised

joemcelroy reviewed (approved):
LGTM apart from the missing localised string.

julianrosado reviewed (approved):
Based on the version you shared in your video to me, it looked good!

![CleanShot 2023-04-25 at 09 05 41](https://user-images.githubusercontent.com/4490333/234285587-708739f2-139d-4475-b774-f73b33cebf72.png)

saarikabhasi commented:
> Based on the version you shared in your video to me, it looked good!
> 

[update] @julianrosado I updated the configuration button to `EuiButtonEmpty` aligning with Design and updated the screen recording. Please let me know if it looks good. And Sorry for creating confusion. 

<img width="1723" alt="Screenshot 2023-04-25 at 9 10 22 AM" src="https://user-images.githubusercontent.com/55930906/234286729-72fd8877-ae1f-4aa8-a203-65545b473b11.png">
